src/patternfly/components/FileUpload/file-upload.scss

```diff
@@ -0,0 +1,132 @@
+    }
+  }
+}
+
+.pf-c-input-group > .pf-c-form-control {
+  &::placeholder {
+    color: var(--pf-c-file-upload--c-input-group--c-form-control--placeholder--Color);
+  }
+}
+
+.pf-c-file-upload__text-area {
+  position: relative;
+  min-height: var(--pf-c-file-upload__text-area--MinHeight);
+  padding: var(--pf-c-file-upload__text-area--PaddingTop) var(--pf-c-file-upload__text-area--PaddingRight) var(--pf-c-file-upload__text-area--PaddingBottom) var(--pf-c-file-upload__text-area--PaddingLeft);
+  resize: vertical;
+  background-color: var(--pf-c-file-upload__text-area--BackgroundColor);
+  border: var(--pf-c-file-upload__text-area--BorderWidth) solid;
+  border-color: var(--pf-c-file-upload__text-area--BorderTopColor) var(--pf-c-file-upload__text-area--BorderRightColor) var(--pf-c-file-upload__text-area--BorderBottomColor) var(--pf-c-file-upload__text-area--BorderLeftColor);
+
+  &:not(:read-only) {
+    &.pf-m-hover,
+    &:hover {
+      --pf-c-file-upload__text-area--BorderBottomColor: var(--pf-c-file-upload__text-area--hover--BorderBottomColor);
+    }
+
+    &.pf-m-active,
+    &:active {
+      --pf-c-file-upload__text-area--BorderBottomColor: var(--pf-c-file-upload__text-area--active--BorderBottomColor);
+
+      border-bottom-width: var(--pf-c-file-upload__text-area--active--BorderBottomWidth);
+    }
+
+    &.pf-m-focus,
+    &:focus {
+      --pf-c-file-upload__text-area--BorderBottomColor: var(--pf-c-file-upload__text-area--focus--BorderBottomColor);
+
+      border-bottom-width: var(--pf-c-file-upload__text-area--focus--BorderBottomWidth);
+    }
+  }
+
+  &:read-only,
+  &.pf-m-readonly {
+    --pf-c-file-upload__text-area--BackgroundColor: var(--pf-c-file-upload__text-area--readonly--BackgroundColor);
+  }
+
+  &.pf-m-loading {
+    background-color: var(--pf-global--disabled-color--300);
+    border: 0;
+  }
+}
+
+.pf-c-button.pf-m-control {
+  --pf-c-button--PaddingRight: var(--pf-c-file-upload--c-button--m-control--PaddingRight);
+  --pf-c-button--PaddingLeft: var(--pf-c-file-upload--c-button--m-control--PaddingLeft);
+
+  outline-offset: var(--pf-c-file-upload--c-button--m-control--OutlineOffset);
+
+  &:disabled,
+  &.pf-m-disabled {
+    color: initial;
+    background-color: var(--pf-c-file-upload--c-button--m-control--disabled--BackgroundColor);
+
+    &::after {
+      content: "";
+      border: var(--pf-c-file-upload--c-button--m-control--disabled--after--BorderWidth) solid;
+      border-color: var(--pf-c-file-upload--c-button--m-control--disabled--after--BorderTopColor) var(--pf-c-file-upload--c-button--m-control--disabled--after--BorderRightColor) var(--pf-c-file-upload--c-button--m-control--disabled--after--BorderBottomColor) var(--pf-c-file-upload--c-button--m-control--disabled--after--BorderLeftColor);
+      border-radius: initial;
+    }
+  }
+}
+
+.pf-c-spinner {
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
```

Comment:
this isn't working because `.pf-c-spinner` also has a default `transform` in it's `animation` that makes it spin. In general I'd try and avoid positioning components like this directly in another component - let's use a wrapper instead to do the positioning, then just drop the component in that. So like `pf-c-form-control__content-spinner/loading` where "content" is whatever you name the main area container. That does the positioning, then you just drop the spinner in it.

Reply:
A trick you can use to create a crosshair element on something is absolutely position the `::before/after` elements as lines that create the crosshair - https://codepen.io/mcoker/pen/yLNygrE

<img width="1457" alt="Screen Shot 2020-02-07 at 7 07 32 PM" src="https://user-images.githubusercontent.com/35148959/74076427-9be07500-49dd-11ea-9345-b0eb63a42e6f.png">

Reply:
ahh thank you!!